# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsHttpChannel.h
# Commit: 02933fde7e3f
# Full Hash: 02933fde7e3f3cf4827730cb670dc76b91022813
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2020-10-13 21:43:51
# Regressor Bug: 1665062
# File Overlap Count: 1
# Description:
#   Bug 1665062: HTTPS-Only: Upgraded website creating HTTP auth prompt gets interrupted by error-page r=necko-reviewers,dragana,JulianWels
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D91908
# ==============================================================================

diff -r bc531943d78a -r 02933fde7e3f netwerk/protocol/http/nsHttpChannel.h
--- a/netwerk/protocol/http/nsHttpChannel.h	Tue Oct 13 18:07:04 2020 +0300
+++ b/netwerk/protocol/http/nsHttpChannel.h	Tue Oct 13 14:06:28 2020 +0000
@@ -152,6 +152,7 @@
   NS_IMETHOD GetEncodedBodySize(uint64_t* aEncodedBodySize) override;
   // nsIHttpChannelInternal
   NS_IMETHOD SetupFallbackChannel(const char* aFallbackKey) override;
+  NS_IMETHOD GetIsAuthChannel(bool* aIsAuthChannel) override;
   NS_IMETHOD SetChannelIsForDownload(bool aChannelIsForDownload) override;
   NS_IMETHOD GetNavigationStartTimeStamp(TimeStamp* aTimeStamp) override;
   NS_IMETHOD SetNavigationStartTimeStamp(TimeStamp aTimeStamp) override;
@@ -652,6 +653,7 @@
   uint32_t mCacheQueueSizeWhenOpen;
 
   Atomic<bool, Relaxed> mCachedContentIsValid;
+  Atomic<bool> mIsAuthChannel;
   Atomic<bool> mAuthRetryPending;
 
   // state flags